ibacm/prov/acmp/src/acmp.c

```diff
@@ -2458,7 +2459,7 @@ static uint16_t acmp_get_pkey_index(struct acm_endpoint *endpoint)
 
 	for (i = 0, ret = 0; !ret; i++) {
 		ret = ibv_query_pkey(port->dev->verbs, port->port_num, i, &pkey);
-		if (!ret && endpoint->pkey == pkey)
+		if (!ret && endpoint->pkey == be16toh(pkey))
```

Comment:
This is the one patch I was most uncertain about.. There are lots of places in acm that store be and le values in the same struct member. However, I could not find any place that set endpoint->pkey to a BE value...